app/controllers/reservations_controller.rb

```diff
@@ -183,6 +183,15 @@ def update
       return
     end
 
+    if !current_user.administrator?
```

Comment:
Yea, you're right about `assign_times_from_params`, I'll fix that.

Also, I'll remove the admin check for the validation. Sorry, I'm still kind of unfamiliar with how each role edits different details, but that makes sense.